### ISSUE TITLE
[2.3] nginx becomes unstable/reloads in an intense way

### ISSUE THREAD

**Issue by lonesomewalker** [bigbluebutton/bigbluebutton#11286]
**Describe the bug**
Unfortunately we are not able to install a dedicated dev version (like alpha4, which was stable) so you always have to use latest.
After installing a default one, the server becomes really unusable by restarting nginx permanently.
This makes it almost impossible to use, since all elements rely on that.

**Additional context**
Install done by bbb-install.sh with -v bionic-230-dev
Is there any chance to chose between which alpha also for the dev branch?

**Comment by ffdixon** [comment]
> After installing a default one, the server becomes really unusable by restarting nginx permanently.

Can you describe in more detail what you are seeing?

**Comment by lonesomewalker** [comment]
Yes, will post logs about that, currently had to replace the machine with 2.2 :-/

But in general:
nginx delivers 404 for almost everything just for seconds, then everything is back to normal.

Means also /b for Greenlight delivers 404

**Comment by rasos** [comment]
I can't confirm this for 2.3 alpha-5. 

We just get sometimes a 404 about a missing  `/var/www/bigbluebutton-default/client/conf/config.xml`

**Comment by cod3r0k** [comment]
> I can't confirm this for 2.3 alpha-5.
> 
> We just get sometimes a 404 about a missing `/var/www/bigbluebutton-default/client/conf/config.xml`

Is there any update? I have the same error log for this phenomenon in 2.3-alpha5 @antobinary

**Comment by cod3r0k** [comment]
> > After installing a default one, the server becomes really unusable by restarting nginx permanently.
> 
> Can you describe in more detail what you are seeing?
Dear @ffdixon 
When user wants to join the session, an error occurs. When you check the /var/log/nginx/error.log , you can see that GET request for  `/var/www/bigbluebutton-default/client/conf/config.xml`. 

I check the issues and found that you wrote some thing about this issue in 2016 (https://groups.google.com/g/bigbluebutton-dev/c/oqflIjikbok/m/1vpZUrzeAwAJ) but it is not really works now for 2.3 dev alpha5.

**Comment by ffdixon** [comment]
Interested if you get any better performance with [alpha-6](https://github.com/bigbluebutton/bigbluebutton/releases/tag/v2.3-alpha-6) (just released), or can provide any more information on the restart of nginx.

**Comment by cod3r0k** [comment]
> Interested if you get any better performance with [alpha-6](https://github.com/bigbluebutton/bigbluebutton/releases/tag/v2.3-alpha-6) (just released), or can provide any more information on the restart of nginx.

Thanks, I will test the new version and send you a feedback.

**Comment by cod3r0k** [comment]
> Interested if you get any better performance with [alpha-6](https://github.com/bigbluebutton/bigbluebutton/releases/tag/v2.3-alpha-6) (just released), or can provide any more information on the restart of nginx.

Dear @ffdixon , Do you know when can we update to newer alpha? because alpha-6 has many crash problem in UI.

**Comment by cod3r0k** [comment]
> or can provide any more information on the restart of nginx.

in alpha-5 I have this problem and it never solved yet.

`2021/02/13 23:31:10 [error] 14019#14019: *23459 open() "/var/www/bigbluebutton-default/client/conf/config.xml" failed (2: No such file or directory), client: xxxxxxxxxxxx, server: xxxxxxxxxxxx, request: "GET /client/conf/config.xml HTTP/1.1", host: "xxxxxxxxxx"`

**Comment by antobinary** [comment]
Config.xml in 2.3+ was addressed via https://github.com/bigbluebutton/bigbluebutton/pull/12432 and https://github.com/bigbluebutton/bigbluebutton/pull/12426

**Comment by antobinary** [comment]
@lonesomewalker have you encountered the "nginx becomes unstable/reloads in an intense way" since (in 2.3.0/ or 2.3.1)?

**Comment by lonesomewalker** [comment]
Unfortunately, come up again today.
config.xml is still searched from BBB2.3.3, so nginx gets busted again.

**Comment by antobinary** [comment]
@lonesomewalker do you see which component tried to reach `config.xml` on your 2.3.3 server?
On 2.4 (alpha1 coming soon) I have removed everything related to config.xml that I could find https://github.com/bigbluebutton/bigbluebutton/pull/12432 But this work has not seen enough testing/usage to even consider backporting to 2.3. 

>will post logs about that

Any chance to send some logs or try to trace back?

**Comment by lonesomewalker** [comment]
Unfortunately i only have the nginx error log which says -> not found -> commented the config related line out -> bbb error because API not available.

Where else should i look to give you proper information?

**Comment by lonesomewalker** [comment]
Okay, after doing ~500 automated setups, i finally found out that for what reason the installer does write:
```bigbluebutton.web.serverURL=https://here.the.ip.address```

I am sure this is wrong :-)

**Comment by antobinary** [comment]
Thank you for narrowing this down!

>`bigbluebutton.web.serverURL=https://here.the.ip.address`

Is this in `/etc/bigbluebutton/bbb-web.properties` ?

**Comment by lonesomewalker** [comment]
Nope, /usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties
